tools/tck-api/src/main/scala/org/opencypher/tools/tck/api/CypherTCK.scala

```diff
@@ -439,6 +440,12 @@ case class ExpectResult(expectedResult: CypherValueRecords, source: io.cucumber.
   }
 }
 
+case class ExpectErrorWithGQLCode(gqlCode: String, source: io.cucumber.core.gherkin.Step) extends Step {
+}
+
+case class ExpectErrorWithGQLCodeAndMessage(gqlCode: String, message: String, source: io.cucumber.core.gherkin.Step) extends Step {
+}
+
 case class ExpectError(errorType: String, phase: String, detail: String, source: io.cucumber.core.gherkin.Step) extends Step {
```

Comment:
May be this can be renamed, too, e.g.:

```suggestion
case class ExpectErrorWithLegacyDetail(errorType: String, phase: String, detail: String, source: io.cucumber.core.gherkin.Step) extends Step {
```

I have a similar suggestion further down.

Reply:
Agreed, easier to differentiate which type of Errors we are dealing with.